roles/addons/nic_nmcli/readme.rst

```diff
@@ -6,27 +6,106 @@ Description
 
 This role configure network interfaces to provide desired ip, prefix, gateway, etc.
 
-This role is still under work, as nmcli is not supported in RHEL 8 for now, and that this role need to handle much more features.
+This role provides all features availables in the main nmcli module:
+https://docs.ansible.com/ansible/latest/collections/community/general/nmcli_module.html
 
 Instructions
 ^^^^^^^^^^^^
 
-NA
+Stack specific behaviors
+""""""""""""""""""""""""
 
-To be done
-^^^^^^^^^^
+While all of the nmcli module options are supported, 
+some provides more integrated features:
+
+* **conn_name**: is equal to **interface**, but has higher precedence over **interface** if both are set.
+* **ifname**: is equal to **physical_device**, but has higher precedence over **ifname**  if both are set.
+* **type**: is set to *ethernet* by default if not set.
+* **ip4**: can be set using a simple ipv4, then role will use **networks[item.network]['.prefix4']** or if not exist to **networks[item.network]['.prefix']** to complete address, or force address with prefix if string *'/'* is present.
+* **ip6**: can be set using a simple ipv6, then role will use **networks[item.network]['.prefix6']** to complete address, or force address with prefix if string *'::'* is present.
+* **mtu**: has higher precedence over **networks[item.network]['mtu']** if both are set.
+* **gw4**: has higher precedence over **networks[item.network]['.gateway4']** which has higher precedence over **networks[item.network]['.gateway']** (if set).
+* **gw6**: has higher precedence over **networks[item.network]['.gateway6']** if both are set.
+
+Basic ipv4
+""""""""""
+
+.. code-block:: yaml
+
+  network_interfaces:
+    - interface: eth0
+      ip4: 10.10.0.1
+      network: ice1-1
+
+Force gateway and MTU
+"""""""""""""""""""""
+
+.. code-block:: yaml
+
+  network_interfaces:
+    - interface: eth0
+      ip4: 10.10.0.1
+      network: ice1-1
+      gw4: 10.10.2.1
+      mtu: 9000
+
+Alias / Multiple ip
+"""""""""""""""""""
+
+In multiple ip modes, you need to set the prefix yourself:
+
+.. code-block:: yaml
 
-Add more features, as nmcli Ansible module can do much more.
+  network_interfaces:
+    - interface: eth0
+      ip4: 10.10.0.1/16,10.10.0.2/16
```

Comment:
could be great to have this as a list instead of coma separated string

Reply:
I will try to do that 🙂

Reply:
Difficult honestly 🙁 We can do that in another PR